src/fparser/common/tests/test_readfortran.py

```diff
@@ -340,6 +340,258 @@ def test_base_free_continuation(log):
+    include_info[include_filename2] = include_code2
+    check_include_works(fortran_filename, fortran_code, include_info,
+                        EXPECTED_CODE, tmpdir)
+
+
+def test_include6(tmpdir, ignore_comments):
+    '''Check that FortranReaderBase can parse an include file correctly
+    when it contains comments. Test with and without comments being
+    ignored.
+
+    '''
+    fortran_filename = "prog.f90"
+    include_filename = "prog.inc"
+    fortran_code = ("program test\n"
+                    "  ! prog comment 1\n"
+                    "  include '{0}'\n"
+                    "  ! prog comment 2\n"
+                    "end program".format(include_filename))
+    include_code = ("! include comment 1\n"
+                    "print *, 'Hello'\n"
+                    "! include comment 2")
+    include_info = {}
+    include_info[include_filename] = include_code
+    if ignore_comments:
+        expected = EXPECTED_CODE
+    else:
+        expected = ("program test\n"
+                    "! prog comment 1\n"
+                    "! include comment 1\n"
+                    "print *, 'Hello'\n"
+                    "! include comment 2\n"
+                    "! prog comment 2\n"
+                    "end program")
+    check_include_works(fortran_filename, fortran_code, include_info,
+                        expected, tmpdir, ignore_comments=ignore_comments)
+
+
+def test_get_item(ignore_comments):
+    '''Check the get_item() function works as expected. Test with and
+    without comments being ignored.
+
+    '''
+    if ignore_comments:
+        expected = EXPECTED_CODE
+    else:
+        expected = ("program test\n"
+                    "! prog comment 1\n"
+                    "print *, 'Hello'\n"
+                    "! prog comment 2\n"
+                    "end program")
+    expected_line = expected.split("\n")
+    reader = FortranStringReader(FORTRAN_CODE, ignore_comments=ignore_comments)
+    while True:
+        output_line = reader.get_item()
+        if not output_line:
+            break
+        assert expected_line.pop(0) in output_line.line
+
+
+def test_put_item(ignore_comments):
+    '''Check that when a line is consumed it can be pushed back so it can
+    be consumed again. Test with and without comments being
+    ignored.
+
+    '''
+    reader = FortranStringReader(FORTRAN_CODE, ignore_comments=ignore_comments)
+    while True:
+        orig_line = reader.get_item()
+        if not orig_line:
+            break
+        reader.put_item(orig_line)
+        fifo_line = reader.get_item()
+        assert fifo_line == orig_line
+
+
+def test_put_item_include(ignore_comments):
+    '''Check that when a line that has been included via an include
+    statement is consumed it can be pushed back so it can be consumed
+    again. Test with and without ignoring comments.
+
+    '''
+    reader = FortranStringReader(FORTRAN_CODE, ignore_comments=ignore_comments)
+    while True:
```

Comment:
As above, although I now realise that `ignore_comments` complicates matters so maybe my suggestion isn't viable.

Reply:
Yes I think comments complicate matters here.